cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -60,28 +61,37 @@ public void classify(CloseableIterator<LogEvent> logIter, HttpResponder responde
       }
       IThrowableProxy throwableProxy = logEvent.getThrowableProxy();
       while (throwableProxy != null) {
-        if (ProgramFailureException.class.getName().equals(throwableProxy.getClassName())) {
-          errorMessage = throwableProxy.getMessage();
+        try {
+          if (FailureDetailsProvider.class.isAssignableFrom(
```

Comment:
nit: Make a small static cache here. Class.forName is expensive

Reply:
done.